tests/test_conftest.py

```diff
@@ -41,6 +45,32 @@ def df_type_from_marker(request: pytest.FixtureRequest) -> bool:
     return "cudf" if use_cudf else "pandas"
 
 
+@pytest.mark.use_cudf
+@pytest.mark.use_pandas
+def test_works_with_marks(dataset: DatasetLoader):
+    # Test is parameterized so df runs twice, once as pandas and another time as cudf
+    df = dataset["filter_probs.csv"]
+    assert isinstance(df, (pd.DataFrame, cudf.DataFrame))
+
+
+def test_only_pandas(dataset_pandas: DatasetLoader):
+    # Test only runs with pandas
+    df = dataset_pandas["filter_probs.csv"]
+    assert isinstance(df, pd.DataFrame)
+
+
+def test_only_cudf(dataset_cudf: DatasetLoader):
+    # Test only runs with cudf
+    df = dataset_cudf["filter_probs.csv"]
+    assert isinstance(df, cudf.DataFrame)
+
+
+def test_both(dataset: DatasetLoader):
+    # By default, requesting dataset will parameterize both
+    df = dataset["filter_probs.csv"]
+    assert isinstance(df, (pd.DataFrame, cudf.DataFrame))
```

Comment:
Prefix these with `test_dataset_XXX` similar to the `test_mark_XXX`, `test_fixture_XXX` and `test_config_XXX`. Makes it easier to distinguish what you are testing.